src/codegen.cpp

```diff
@@ -4865,6 +4856,21 @@ static GlobalVariable *global_to_llvm(const std::string &cname, void *addr, Modu
     add_named_global(gv, addr);
     return gv;
 }
+std::map<jl_value_t**, GlobalVariable*> gv_for_global;
+static GlobalVariable *global_to_llvm(const std::string &cname, jl_value_t **addr, Module *m)
+{
+    GlobalVariable *gv = global_to_llvm(cname, (void*)addr, m);
+    gv_for_global[addr] = gv;
+    return gv;
+}
+static GlobalVariable *julia_const_gv(jl_value_t *val)
+{
+    for (auto& kv : gv_for_global) {
+        if (*kv.first == val)
+            return kv.second;
+    }
```

Comment:
At the point of populating the map, values like `jl_inexact_exception` aren't known yet (they are defined in `boot.jl`).